From c14b6cb1d0df5dfa9a17e29032f9347f7a6f0b05 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@freescale.com>
Date: Mon, 11 Jun 2012 15:04:49 -0500
Subject: [PATCH 290/502] ENGR00181314 [Display] - Enhance support for second
 display.

Change the default pixel format for the second display to 32bpp.

Signed-off-by: Oliver Brown <oliver.brown@freescale.com>
---
 mx5x/libgralloc/framebuffer.cpp | 24 ++++++++----------------
 1 file changed, 8 insertions(+), 16 deletions(-)

diff --git a/mx5x/libgralloc/framebuffer.cpp b/mx5x/libgralloc/framebuffer.cpp
index be3000e..6eb18f3 100755
--- a/mx5x/libgralloc/framebuffer.cpp
+++ b/mx5x/libgralloc/framebuffer.cpp
@@ -1216,20 +1216,7 @@ static int mapSecFrameBuffer(fb_context_t* ctx)
     info.xoffset = 0;
     info.yoffset = 0;
     info.activate = FB_ACTIVATE_NOW;
-                
-    /*
-    * Explicitly request 5/6/5
-    */
-    info.bits_per_pixel = 16;
-    info.nonstd = 0;
-    info.red.offset     = 11;
-    info.red.length     = 5;
-    info.green.offset   = 5;
-    info.green.length   = 6;
-    info.blue.offset    = 0;
-    info.blue.length    = 5;
-    info.transp.offset  = 0;
-    info.transp.length  = 0;
+
     info.yres_virtual = ALIGN_PIXEL_128(info.yres) * nr_framebuffers;
     info.xres_virtual = ALIGN_PIXEL(info.xres);
                         
@@ -1362,7 +1349,12 @@ static int resizeToSecFrameBuffer(int base,int phys,fb_context_t* ctx)
     ctx->mTask.input.crop.h = ctx->device.height;
     ctx->mTask.input.paddr = phys;
 
-    ctx->mTask.output.format = v4l2_fourcc('R', 'G', 'B', 'P');
+    if (ctx->sec_info.bits_per_pixel == 32) {
+        ctx->mTask.output.format = v4l2_fourcc('B', 'G', 'R', '4');
+    }
+    else {
+        ctx->mTask.output.format = v4l2_fourcc('R', 'G', 'B', 'P');
+    }
     ctx->mTask.output.width = ctx->sec_disp_w;
     ctx->mTask.output.height = ctx->sec_disp_h;
     ctx->mTask.output.crop.w = ctx->sec_disp_w;
@@ -1410,7 +1402,7 @@ static int resizeToSecFrameBuffer(int base,int phys,fb_context_t* ctx)
                 ctx->mTask.output.crop.w -= 8;
                 break;
             case IPU_CHECK_ERR_SPLIT_OUTPUTH_OVER:
-                ctx->mTask.output.crop.h -= 8;;
+                ctx->mTask.output.crop.h -= 8;
                 break;
             default:
                 return status;
-- 
1.8.0

